### PR TITLE
Fixed some dynamic bugs

### LINKED ISSUE #47734: Dynamic only looks for enemy roles among players eligible for playing as the antag
## Reproduction:

```for (var/mob/M in living_players)```

```living_players = trim_list(mode.current_players[CURRENT_LIVING_PLAYERS])```

trim_list removes all players from the list who: do not meet the required age **for the antag**, are jobbanned **from the antag**, have a role that is restricted **from the antag** (!!!!!! prevents most antags that require sec enemies from working completely!!) etc.

i'd make a PR but my local repo is busted due to being a defunct downstream and i can't particularly be assed to fix it

### LINKED ISSUE #47756: [DYNAMIC] non-repeatable midround ruletypes repeat anyway (exact fix within)
```
	if(!rule.repeatable)
		if(rule.ruletype == "Latejoin")
			latejoin_rules = remove_from_list(latejoin_rules, rule.type)
		else if(rule.type == "Midround")
			midround_rules = remove_from_list(midround_rules, rule.type)
```
note
```
else if(rule.type == "Midround")
```
should be
```
else if(rule.ruletype == "Midround")
```

### PR DESCRIPTION
**Pull request by kriskog**
## About The Pull Request
Fixes a couple bugs reported by @Putnam3145 

## Why It's Good For The Game
Fixes #47734
Fixes #47756

## Changelog
:cl: Skoglol
fix: Dynamic midround rulesets should now respect the repeatable setting.
fix: Dynamic midround/latejoin rulesets now look for enemy roles in the right place.
/:cl:

### CODE REVIEW

File: config/admins.txt [hunk 1]
```diff
@@ -137,3 +137,4 @@ shizcalev = Game Master
 NicBR = Game Master
 LoserWasTaken = Game Master
 Fikou = Game Master
+Skoglol = Game Master
```
Comment (bobbahbrown) [inline config/admins.txt]: that's not a bug...

Comment (Ziiro) [reply config/admins.txt]: 🤔

Comment (kriskog) [reply config/admins.txt]: :thinking:

**Comment by ShizCalev** [comment]
would you mind making your changelog a little less vague?

**Comment by kriskog** [comment]
Done

**Review by TheChosenEvilOne — APPROVED** [review]
I as the dynamic man approve this PR.

**Pull request merged**